pkg/operator/controller/garden/reconciler.go

```diff
@@ -59,24 +63,60 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
+	conditionReconciled := v1beta1helper.GetOrInitConditionWithClock(r.Clock, garden.Status.Conditions, operatorv1alpha1.GardenReconciled)
+	conditionReconciled = v1beta1helper.UpdatedConditionWithClock(r.Clock, conditionReconciled, gardencorev1beta1.ConditionProgressing, conditionReasonPrefix(garden)+"Progressing", "Garden operation is currently being processed.")
+	if err := r.patchConditions(ctx, garden, conditionReconciled); err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not patch status of %s condition to %s: %w", conditionReconciled.Type, conditionReconciled.Status, err)
+	}
+
 	secretsManager, err := secretsmanager.New(
 		ctx,
 		log.WithName("secretsmanager"),
-		clock.RealClock{},
+		r.Clock,
 		r.RuntimeClient,
 		r.GardenNamespace,
 		operatorv1alpha1.SecretManagerIdentityOperator,
 		secretsmanager.Config{CASecretAutoRotation: true},
 	)
 	if err != nil {
-		return reconcile.Result{}, err
+		return reconcile.Result{}, r.patchConditionToFalse(ctx, log, garden, conditionReconciled, err)
 	}
 
 	if garden.DeletionTimestamp != nil {
-		return r.delete(ctx, log, garden, secretsManager)
+		if result, err := r.delete(ctx, log, garden, secretsManager); err != nil {
+			return result, r.patchConditionToFalse(ctx, log, garden, conditionReconciled, err)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if result, err := r.reconcile(ctx, log, garden, secretsManager); err != nil {
+		return result, r.patchConditionToFalse(ctx, log, garden, conditionReconciled, err)
+	}
+
+	patch := client.MergeFrom(garden.DeepCopy())
+	garden.Status.Conditions = v1beta1helper.MergeConditions(garden.Status.Conditions, v1beta1helper.UpdatedConditionWithClock(r.Clock, conditionReconciled, gardencorev1beta1.ConditionTrue, conditionReasonPrefix(garden)+"Successful", "Garden operation was completed successfully."))
```

Comment:
We should have a `status.gardener` field similar to the shoot status which includes the gardener-operator's version.